AppHandling/Start-NavContainerAppDataUpgrade.ps1

```diff
@@ -18,39 +18,59 @@ function  Start-BcContainerAppDataUpgrade {
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string] $appName,
         [string] $appVersion = "",
-        [string] $language = ""
+        [string] $language = "",
+        [string] $exclusiveAccessTicket = "",
+        [switch] $force,
+        [switch] $skipVersionCheck,
+        [ValidateSet("Add", "Clean", "Development", "ForceSync", "None")]
+        [string] $syncMode
     )
 
-$telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
-try {
+    $telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
+    try {
 
-    Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param($appName, $appVersion, $tenant, $language)
-        Write-Host "Upgrading app $appName"
-        $parameters = @{
-            "ServerInstance" = $ServerInstance;
-            "Name" = $appName;
-            "Tenant" = $tenant
-        }
-        if ($appVersion) {
-            $parameters += @{ "Version" = $appVersion }
-        }
-        if ($language) {
-            $parameters += @{ "Language" = $language }
-        }
-        Start-NAVAppDataUpgrade @parameters
-    } -ArgumentList $appName, $appVersion, $tenant, $language
-    Write-Host -ForegroundColor Green "App successfully upgraded"
-}
-catch {
-    TrackException -telemetryScope $telemetryScope -errorRecord $_
-    throw
-}
-finally {
-    TrackTrace -telemetryScope $telemetryScope
-}
+        Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param($appName, $appVersion, $tenant, $language, $exclusiveAccessTicket, $path, $syncMode, $force, $skipVersionCheck)
+            Write-Host "Upgrading app $appName"
+            $parameters = @{
+                "ServerInstance" = $ServerInstance;
+                "Name"           = $appName;
+                "Tenant"         = $tenant
+            }
+            if ($appVersion) {
+                $parameters += @{ "Version" = $appVersion }
+            }
+            if ($language) {
+                $parameters += @{ "Language" = $language }
+            }
+            if ($exclusiveAccessTicket) {
+                $parameters += @{ "ExclusiveAccessTicket" = $exclusiveAccessTicket }
+            }
+            if ($path) {
+                $parameters += @{ "Path" = $path }
+            }
+            if ($syncMode) {
+                $parameters += @{ "SyncMode" = $syncMode }
+            }
+            if ($force.IsPresent) {
+                $parameters += @{ "Force" = $force }
+            }
+            if ($skipVersionCheck.IsPresent) {
+                $parameters += @{ "SkipVersionCheck" = $skipVersionCheck }
+            }
+            Start-NAVAppDataUpgrade @parameters
+        } -ArgumentList $appName, $appVersion, $tenant, $language, $exclusiveAccessTicket, $path, $syncMode, $force, $skipVersionCheck

```

Comment:
path?

Reply:
I was surprised that I can set path, but yes. This is from documentation.
Start-NAVAppDataUpgrade
     [-ServerInstance] <String>
     [-Tenant <TenantId>]
     -Path <String>
     [-Language <LanguageSetting>]
     [-ExclusiveAccessTicket <String>]
     [-PassThru]
     [-SkipVersionCheck]
     [-SyncMode <NavAppSyncMode>]
     [-Force]
     [<CommonParameters>]